Subject: [patch] uml allow vsyscall code to build on 24.patch
From: Bodo Stroesser <bstroesser@fujitsu-siemens.com>
Patch-mainline: 2.6.11-rc1

This patch fixes compilation on 2.4 hosts by not relying on macros from 2.6
host kernel headers in one userspace file. It's about AT_SYSINFO_* macros.

Signed-off-by: Paolo 'Blaisorblade' Giarrusso <blaisorblade_spam@yahoo.it>
Signed-off-by: Jeff Dike <jdike@addtoit.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Acked-by: Gerd Knorr <kraxel@suse.de>
---
 arch/um/include/elf_user.h |   19 +++++++++++++++++++
 arch/um/os-Linux/elf_aux.c |    1 +
 2 files changed, 20 insertions(+)

diff -puN /dev/null arch/um/include/elf_user.h
--- /dev/null	Thu Apr 11 07:25:15 2002
+++ 25-akpm/arch/um/include/elf_user.h	Fri Dec  3 13:50:38 2004
@@ -0,0 +1,19 @@
+/*
+ * Copyright (C) 2004 Fujitsu Siemens Computers GmbH
+ * Author: Bodo Stroesser <bstroesser@fujitsu-siemens.com>
+ * Licensed under the GPL
+ */
+
+#ifndef __ELF_USER_H__
+#define __ELF_USER_H__
+
+/* For compilation on a host that doesn't support AT_SYSINFO (Linux 2.4)  */
+
+#ifndef AT_SYSINFO
+#define AT_SYSINFO 32
+#endif
+#ifndef AT_SYSINFO_EHDR
+#define AT_SYSINFO_EHDR 33
+#endif
+
+#endif
diff -puN arch/um/os-Linux/elf_aux.c~uml-allow-vsyscall-code-to-build-on-24 arch/um/os-Linux/elf_aux.c
--- 25/arch/um/os-Linux/elf_aux.c~uml-allow-vsyscall-code-to-build-on-24	Fri Dec  3 13:50:38 2004
+++ 25-akpm/arch/um/os-Linux/elf_aux.c	Fri Dec  3 13:50:38 2004
@@ -10,6 +10,7 @@
 #include <elf.h>
 #include <stddef.h>
 #include "init.h"
+#include "elf_user.h"
 
 #if ELF_CLASS == ELFCLASS32
 typedef Elf32_auxv_t elf_auxv_t;
_
